File: app/src/main/java/com/codepath/debuggingchallenges/models/Movie.java
package com.codepath.debuggingchallenges.models;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class Movie {
    private String title;
    private String posterUrl;
    private double rating;

    public Movie(JSONObject jsonObject) throws JSONException {
        this.posterUrl = jsonObject.getString("poster_path");
        this.title = jsonObject.getString("original-title");
        this.rating = jsonObject.getDouble("vote_average");
    }

    public String getTitle() {
        return title;
    }

    public double getRating() {
        return rating;
    }

    public String getPosterUrl() {
        return String.format("https://image.tmdb.org/t/p/w342/%s", posterUrl);
    }

    public static ArrayList<Movie> fromJSONArray(JSONArray jsonArray) {
        ArrayList<Movie> results = new ArrayList<>();
        for (int i = 0; i < jsonArray.length(); i++) {
            try {
                results.add(new Movie(jsonArray.getJSONObject(i)));
            } catch (JSONException e) {
                e.printStackTrace();
            }
        }
        return results;
    }
}

File: app/src/main/java/com/codepath/debuggingchallenges/models/Movie.kt
package com.codepath.debuggingchallenges.models

import org.json.JSONObject
import org.json.JSONArray
import org.json.JSONException
import java.util.ArrayList

class Movie(jsonObject: JSONObject) {
    val title: String = jsonObject.getString("original_title")
    val posterUrl: String = String.format("https://image.tmdb.org/t/p/w342/%s", jsonObject.getString("poster_path"))
    val rating: Double = jsonObject.getDouble("vote_average")

    companion object {
        fun fromJSONArray(jsonArray: JSONArray): ArrayList<Movie> {
            val results = ArrayList<Movie>()
            for (i in 0 until jsonArray.length()) {
                try {
                    results.add(Movie(jsonArray.getJSONObject(i)))
                } catch (e: JSONException) {
                    e.printStackTrace()
                }
            }
            return results
        }
    }
}